connector/signaltometricsconnector/factory.go

```diff
@@ -32,26 +38,83 @@ func createDefaultConfig() component.Config {
 func createTracesToMetrics(
 	_ context.Context,
 	set connector.Settings,
-	_ component.Config,
+	cfg component.Config,
 	nextConsumer consumer.Metrics,
 ) (connector.Traces, error) {
-	return newSignalToMetrics(set, nextConsumer), nil
+	c := cfg.(*config.Config)
+	parser, err := ottlspan.NewParser(customottl.SpanFuncs(), set.TelemetrySettings)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OTTL statement parser for datapoints: %w", err)
+	}
+
+	metricDefs := make([]model.MetricDef[ottlspan.TransformContext], 0, len(c.Spans))
+	for _, info := range c.Spans {
+		var md model.MetricDef[ottlspan.TransformContext]
+		if err := md.FromMetricInfo(info, parser, set.TelemetrySettings); err != nil {
+			return nil, fmt.Errorf("failed to parse provided metric information; %w", err)
+		}
+		metricDefs = append(metricDefs, md)
+	}
+
+	return &signalToMetrics{
+		logger:         set.Logger,
+		next:           nextConsumer,
+		spanMetricDefs: metricDefs,
+	}, nil
 }
 
 func createMetricsToMetrics(
 	_ context.Context,
 	set connector.Settings,
-	_ component.Config,
+	cfg component.Config,
 	nextConsumer consumer.Metrics,
 ) (connector.Metrics, error) {
-	return newSignalToMetrics(set, nextConsumer), nil
+	c := cfg.(*config.Config)
+	parser, err := ottldatapoint.NewParser(customottl.DatapointFuncs(), set.TelemetrySettings)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OTTL statement parser for datapoints: %w", err)
+	}
+
+	metricDefs := make([]model.MetricDef[ottldatapoint.TransformContext], 0, len(c.Datapoints))
+	for _, info := range c.Datapoints {
+		var md model.MetricDef[ottldatapoint.TransformContext]
+		if err := md.FromMetricInfo(info, parser, set.TelemetrySettings); err != nil {
+			return nil, fmt.Errorf("failed to parse provided metric information; %w", err)
+		}
+		metricDefs = append(metricDefs, md)
+	}
+
+	return &signalToMetrics{
+		logger:       set.Logger,
+		next:         nextConsumer,
+		dpMetricDefs: metricDefs,
+	}, nil
 }
 
 func createLogsToMetrics(
 	_ context.Context,
 	set connector.Settings,
-	_ component.Config,
+	cfg component.Config,
 	nextConsumer consumer.Metrics,
 ) (connector.Logs, error) {
-	return newSignalToMetrics(set, nextConsumer), nil
+	c := cfg.(*config.Config)
+	parser, err := ottllog.NewParser(customottl.LogFuncs(), set.TelemetrySettings)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OTTL statement parser for datapoints: %w", err)
```

Comment:
Is `"datapoints"` accurate here or it should be `"logs"`?
```suggestion
		return nil, fmt.Errorf("failed to create OTTL statement parser for logs: %w", err)
```